components/keyrings/common/data/data.h

```diff
@@ -24,13 +24,58 @@
 #define DATA_INCLUDED
 
 #include <string>
+#include "global_default_mr.h"
 
 namespace keyring_common {
 namespace data {
 
 /** Data types */
-using Type = std::string;
-using Sensitive_data = std::string;
+using Type = pmr_string;
+//using Sensitive_data = pmr_string;
+
+struct Sensitive_data {
+
+  Sensitive_data(){}
+
+  Sensitive_data(pmr_string const& str) : data(str) {
+    encode();
+  }
+
+  Sensitive_data(const char* str) : data(str) {
+    encode();
+  }
+  Sensitive_data(const char* str, std::size_t len) : data(str, len) {
+    encode();
+  }
+
+  std::size_t size() const { return data.size(); }
+  std::size_t length() const { return data.size(); }
+
+  pmr_string decode() const { 
+    auto ret = data;
+    for(auto& c: ret) {
+      // TODO: encode / decode with some actual key, e.g. memory address
+      c ^= 11;
+    }
+    return ret;
+  }
+
+  friend inline bool operator==(Sensitive_data const &a,
```

Comment:
The function is implicitly `inline` already because its body is defined within class definition.